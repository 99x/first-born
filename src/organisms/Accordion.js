import React, { Component } from "react";
import { FlatList } from "react-native";
import PropTypes from "prop-types";
import { AccordionElement } from "../molecules/cards/AccordionElement";

export class Accordion extends Component {
    render() {
        const { horizontal, renderItem, ...otherProps } = this.props;

        return (
            <FlatList
                keyExtractor={this.keyExtractor}
                renderItem={
                    renderItem
                        ? renderItem
                        : ({ item }) => {
                              return (
                                  <AccordionElement
                                      {...item}
                                      isList
                                      {...otherProps}
                                  />
                              );
                          }
                }
                {...otherProps}
            />
        );
    }

    keyExtractor = () => (Math.floor(Math.random() * 1000000) + 1).toString();
}

Accordion.propTypes = {
    bodyColor: PropTypes.string,
    headerColor: PropTypes.string,
    expandedIcon: PropTypes.string,
    icon: PropTypes.string,
    ...FlatList.propTypes
};
