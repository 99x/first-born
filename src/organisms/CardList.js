import React, { Component } from "react";
import { FlatList, Dimensions, View, Platform } from "react-native";
import { Card } from "../molecules/cards/Card";

const { width } = Dimensions.get("window");

export class CardList extends Component {

    render() {
        const { backgroundColor, horizontal, margin, ...otherProps } = this.props;

        return (
            <FlatList
                style={{ marginTop: 5 }}
                keyExtractor={this.keyExtractor}
                renderItem={({ item }) => {
                    return (
                        <View style={{ marginRight: margin }}>
                            <Card image={item.image} title={item.title} description={item.description} backgroundColor={backgroundColor} style={{ width: width - 40 }} />
                        </View>
                    )
                }}
                horizontal
                {...otherProps} />
        )
    }

    keyExtractor = () => (Math.floor(Math.random() * 1000000) + 1).toString();
}

CardList.propTypes = {
    ...FlatList.propTypes
}

CardList.defaultProps = {
    margin: Platform.OS === "ios" ? 15 : 10
}
