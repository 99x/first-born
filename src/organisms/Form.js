import React, { Component } from "react";
import { View, StyleSheet } from "react-native";
import PropTypes from "prop-types";

import { Picker } from "../atoms/Picker";

import { FormDatePicker } from "../molecules/formElements/FormDatePicker";
import { FormInput } from "../molecules/formElements/FormInput";
import { FormPicker } from "../molecules/formElements/FormPicker";
import { FormTextArea } from "../molecules/formElements/FormTextArea";

export class Form extends Component {
    render() {
        const { color, formElements, ...otherProps } = this.props;

        const inputColor = color ? color : "#486c86";

        return (
            <View style={styles.container}>
                {formElements.map((element, key) => {
                    return (
                        <View style={styles.formElementContainer} key={key}>
                            {this.renderElements(element, inputColor, otherProps)}
                        </View>
                    );
                })}
            </View>
        );
    }

    renderElements(element, color, styleProps) {
        element.color = color;
        const { type, pickerData, ...otherProps } = element;

        switch (type) {
            case "text":
                return <FormInput {...styleProps} {...otherProps} />;
            case "textarea":
                return <FormTextArea {...styleProps} {...otherProps} />;
            case "date":
                return <FormDatePicker {...styleProps} {...otherProps} />;
            case "picker":
                return (
                    <FormPicker {...styleProps} {...otherProps}>
                        {pickerData.map((dataElement, key) => (
                            <Picker.Item {...dataElement} key={key} />
                        ))}
                    </FormPicker>
                );
        }
    }
}

Form.propTypes = {
    color: PropTypes.string,
    activeStyle: PropTypes.object,
    errorStyle: PropTypes.object,
    rounded: PropTypes.bool,
    underline: PropTypes.bool,
    defaultStyle: PropTypes.bool,
    noStyle: PropTypes.bool,
    errorColor: PropTypes.string,
};

const styles = StyleSheet.create({
    container: {
        flex: 1,
        width: "100%"
    },
    formElementContainer: {
        marginTop: 10
    }
});
