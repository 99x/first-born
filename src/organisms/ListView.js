import React, { Component } from "react";
import { FlatList } from "react-native";
import PropTypes from "prop-types";
import { ListItem } from "../molecules/listItem/ListItem";
import { ThinListItem } from "../molecules/listItem/ThinListItem";

export class ListView extends Component {
    render() {
        const {
            backgroundColor,
            horizontal,
            renderItem,
            secondary,
            thin,
            rounded,
            ...otherProps
        } = this.props;

        return (
            <FlatList
                style={{ width: "100%" }}
                keyExtractor={this.keyExtractor}
                renderItem={
                    renderItem
                        ? renderItem
                        : ({ item }) => {
                              if (thin) {
                                  return (
                                      <ThinListItem
                                          {...item}
                                          backgroundColor={backgroundColor}
                                          secondary={secondary}
                                          rounded={rounded}
                                      />
                                  );
                              }
                              return (
                                  <ListItem
                                      {...item}
                                      backgroundColor={backgroundColor}
                                      secondary={secondary}
                                      rounded={rounded}
                                  />
                              );
                          }
                }
                {...otherProps}
            />
        );
    }

    keyExtractor = () => (Math.floor(Math.random() * 1000000) + 1).toString();
}

ListView.propTypes = {
    backgroundColor: PropTypes.string,
    thin: PropTypes.bool,
    rounded: PropTypes.bool,
    ...FlatList.propTypes
};

ListView.defaultProps = {
    thin: false,
    rounded: true
};
