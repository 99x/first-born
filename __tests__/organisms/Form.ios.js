import "react-native";
import React from "react";
import renderer from "react-test-renderer";
import { Form } from "../../src/organisms/Form";

jest.mock("Platform", () => {
    const Platform = require.requireActual("Platform");
    Platform.OS = "ios";
    return Platform;
});

it("renders a form with input types text", () => {
    const formElements = [
        { label: "Full Name", type: "text", placeholder: "John Doe" }
    ];

    const tree = renderer.create(<Form formElements={formElements} />).toJSON();
    expect(tree).toMatchSnapshot();
});

it("renders a form with input types, text and textarea", () => {
    const formElements = [
        { label: "Full Name", type: "text", placeholder: "John Doe" },
        { label: "Address", type: "textarea" }
    ];

    const tree = renderer.create(<Form formElements={formElements} />).toJSON();
    expect(tree).toMatchSnapshot();
});

it("renders a form with input types, text, textarea and picker", () => {
    const pickerData = [
        { value: "1", label: "1" },
        { value: "2", label: "2" },
        { value: "3", label: "3" }
    ];

    const formElements = [
        { label: "Full Name", type: "text", placeholder: "John Doe" },
        { label: "Type", type: "picker", pickerData: pickerData },
        { label: "Address", type: "textarea" }
    ];

    const tree = renderer.create(<Form formElements={formElements} />).toJSON();
    expect(tree).toMatchSnapshot();
});

it("renders a form with all input types", () => {
    const pickerData = [
        { value: "1", label: "1" },
        { value: "2", label: "2" },
        { value: "3", label: "3" }
    ];

    const formElements = [
        { label: "Full Name", type: "text", placeholder: "John Doe" },
        { label: "Type", type: "picker", pickerData: pickerData },
        { label: "Date", type: "date", defaultDate: new Date(1554104240475) },
        { label: "Address", type: "textarea" }
    ];

    const tree = renderer.create(<Form formElements={formElements} />).toJSON();
    expect(tree).toMatchSnapshot();
});
