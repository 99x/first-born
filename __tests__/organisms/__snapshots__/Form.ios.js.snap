// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a form with all input types 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "marginTop": 10,
      }
    }
  >
    <View
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgba(33, 33, 33, 0.87)",
            "fontSize": 15,
            "fontWeight": "normal",
            "marginVertical": 5,
            "textAlign": "left",
          }
        }
      >
        Full Name
      </Text>
      <View
        style={
          Object {
            "borderColor": "rgba(33, 33, 33, 0.5)",
            "borderRadius": 15,
            "borderWidth": 0.9,
            "height": 45,
            "marginVertical": 10,
            "paddingHorizontal": 10,
            "width": "100%",
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          onSubmitEditing={[Function]}
          placeholder="John Doe"
          placeholderTextColor="rgba(33, 33, 33, 0.5)"
          style={
            Object {
              "height": "100%",
              "textAlignVertical": "bottom",
              "width": "100%",
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "marginTop": 10,
      }
    }
  >
    <View
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgba(33, 33, 33, 0.87)",
            "fontSize": 15,
            "fontWeight": "normal",
            "marginVertical": 5,
            "textAlign": "left",
          }
        }
      >
        Type
      </Text>
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "borderColor": "rgba(33, 33, 33, 0.5)",
              "borderRadius": 15,
              "borderWidth": 0.9,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "center",
              "marginVertical": 10,
              "paddingLeft": 10,
              "width": "100%",
            }
          }
        >
          <Text
            onPress={[Function]}
            style={
              Object {
                "color": "rgba(33, 33, 33, 0.5)",
                "flex": 1,
                "fontSize": 16,
                "fontWeight": "normal",
                "textAlign": "left",
              }
            }
          >
            Select Option
          </Text>
          <Text
            allowFontScaling={false}
            onPress={[Function]}
            style={
              Array [
                Object {
                  "color": "rgba(33, 33, 33, 0.5)",
                  "fontSize": 24,
                },
                Object {
                  "paddingRight": 15,
                },
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
        <Modal
          animationType="fade"
          hardwareAccelerated={false}
          onRequestClose={[Function]}
          supportedOrientations={
            Array [
              "portrait",
              "landscape",
            ]
          }
          transparent={true}
          visible={false}
        >
          <Text
            onPress={[Function]}
            style={
              Object {
                "color": "#000",
                "flex": 1,
                "fontSize": 16,
                "fontWeight": "normal",
                "textAlign": "left",
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "#FFF",
                "flex": 1,
              }
            }
          >
            <RCTPicker
              items={
                Array [
                  Object {
                    "label": "1",
                    "textColor": undefined,
                    "value": "1",
                  },
                  Object {
                    "label": "2",
                    "textColor": undefined,
                    "value": "2",
                  },
                  Object {
                    "label": "3",
                    "textColor": undefined,
                    "value": "3",
                  },
                ]
              }
              onChange={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              selectedIndex={0}
              style={
                Array [
                  Object {
                    "height": 216,
                  },
                  undefined,
                ]
              }
            />
          </View>
          <Text
            onPress={[Function]}
            style={
              Object {
                "color": "#000",
                "flex": 1,
                "fontSize": 16,
                "fontWeight": "normal",
                "textAlign": "left",
              }
            }
          />
        </Modal>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "marginTop": 10,
      }
    }
  >
    <View
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgba(33, 33, 33, 0.87)",
            "fontSize": 15,
            "fontWeight": "normal",
            "marginVertical": 5,
            "textAlign": "left",
          }
        }
      >
        Date
      </Text>
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "borderColor": "rgba(33, 33, 33, 0.5)",
              "borderRadius": 15,
              "borderWidth": 0.9,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "center",
              "marginVertical": 10,
              "paddingLeft": 10,
              "width": "100%",
            }
          }
        >
          <Text
            onPress={[Function]}
            style={
              Object {
                "color": "rgba(33, 33, 33, 0.5)",
                "flex": 1,
                "fontSize": 16,
                "fontWeight": "normal",
                "textAlign": "left",
              }
            }
          >
            Select Date
          </Text>
          <Text
            allowFontScaling={false}
            onPress={[Function]}
            style={
              Array [
                Object {
                  "color": "rgba(33, 33, 33, 0.5)",
                  "fontSize": 20,
                },
                Object {
                  "paddingRight": 15,
                },
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
        <Modal
          animationType="fade"
          hardwareAccelerated={false}
          onRequestClose={[Function]}
          supportedOrientations={
            Array [
              "portrait",
              "landscape",
            ]
          }
          transparent={true}
          visible={false}
        >
          <Text
            onPress={[Function]}
            style={
              Object {
                "color": "#000",
                "flex": 1,
                "fontSize": 16,
                "fontWeight": "normal",
                "textAlign": "left",
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "#FFF",
                "flex": 1,
              }
            }
          >
            <RCTDatePicker
              date={1554104240475}
              mode="date"
              onChange={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "height": 216,
                }
              }
            />
          </View>
          <Text
            onPress={[Function]}
            style={
              Object {
                "color": "#000",
                "flex": 1,
                "fontSize": 16,
                "fontWeight": "normal",
                "textAlign": "left",
              }
            }
          />
        </Modal>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "marginTop": 10,
      }
    }
  >
    <View
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgba(33, 33, 33, 0.87)",
            "fontSize": 15,
            "fontWeight": "normal",
            "marginVertical": 5,
            "textAlign": "left",
          }
        }
      >
        Address
      </Text>
      <View
        style={
          Object {
            "borderColor": "rgba(33, 33, 33, 0.5)",
            "borderRadius": 15,
            "borderWidth": 0.9,
            "marginVertical": 10,
            "paddingHorizontal": 10,
            "paddingTop": 5,
            "width": "100%",
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          multiline={true}
          onBlur={[Function]}
          onContentSizeChange={[Function]}
          onFocus={[Function]}
          style={
            Object {
              "height": 100,
              "width": "100%",
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`renders a form with input types text 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "marginTop": 10,
      }
    }
  >
    <View
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgba(33, 33, 33, 0.87)",
            "fontSize": 15,
            "fontWeight": "normal",
            "marginVertical": 5,
            "textAlign": "left",
          }
        }
      >
        Full Name
      </Text>
      <View
        style={
          Object {
            "borderColor": "rgba(33, 33, 33, 0.5)",
            "borderRadius": 15,
            "borderWidth": 0.9,
            "height": 45,
            "marginVertical": 10,
            "paddingHorizontal": 10,
            "width": "100%",
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          onSubmitEditing={[Function]}
          placeholder="John Doe"
          placeholderTextColor="rgba(33, 33, 33, 0.5)"
          style={
            Object {
              "height": "100%",
              "textAlignVertical": "bottom",
              "width": "100%",
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`renders a form with input types, text and textarea 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "marginTop": 10,
      }
    }
  >
    <View
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgba(33, 33, 33, 0.87)",
            "fontSize": 15,
            "fontWeight": "normal",
            "marginVertical": 5,
            "textAlign": "left",
          }
        }
      >
        Full Name
      </Text>
      <View
        style={
          Object {
            "borderColor": "rgba(33, 33, 33, 0.5)",
            "borderRadius": 15,
            "borderWidth": 0.9,
            "height": 45,
            "marginVertical": 10,
            "paddingHorizontal": 10,
            "width": "100%",
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          onSubmitEditing={[Function]}
          placeholder="John Doe"
          placeholderTextColor="rgba(33, 33, 33, 0.5)"
          style={
            Object {
              "height": "100%",
              "textAlignVertical": "bottom",
              "width": "100%",
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "marginTop": 10,
      }
    }
  >
    <View
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgba(33, 33, 33, 0.87)",
            "fontSize": 15,
            "fontWeight": "normal",
            "marginVertical": 5,
            "textAlign": "left",
          }
        }
      >
        Address
      </Text>
      <View
        style={
          Object {
            "borderColor": "rgba(33, 33, 33, 0.5)",
            "borderRadius": 15,
            "borderWidth": 0.9,
            "marginVertical": 10,
            "paddingHorizontal": 10,
            "paddingTop": 5,
            "width": "100%",
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          multiline={true}
          onBlur={[Function]}
          onContentSizeChange={[Function]}
          onFocus={[Function]}
          style={
            Object {
              "height": 100,
              "width": "100%",
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`renders a form with input types, text, textarea and picker 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "marginTop": 10,
      }
    }
  >
    <View
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgba(33, 33, 33, 0.87)",
            "fontSize": 15,
            "fontWeight": "normal",
            "marginVertical": 5,
            "textAlign": "left",
          }
        }
      >
        Full Name
      </Text>
      <View
        style={
          Object {
            "borderColor": "rgba(33, 33, 33, 0.5)",
            "borderRadius": 15,
            "borderWidth": 0.9,
            "height": 45,
            "marginVertical": 10,
            "paddingHorizontal": 10,
            "width": "100%",
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          onSubmitEditing={[Function]}
          placeholder="John Doe"
          placeholderTextColor="rgba(33, 33, 33, 0.5)"
          style={
            Object {
              "height": "100%",
              "textAlignVertical": "bottom",
              "width": "100%",
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "marginTop": 10,
      }
    }
  >
    <View
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgba(33, 33, 33, 0.87)",
            "fontSize": 15,
            "fontWeight": "normal",
            "marginVertical": 5,
            "textAlign": "left",
          }
        }
      >
        Type
      </Text>
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "borderColor": "rgba(33, 33, 33, 0.5)",
              "borderRadius": 15,
              "borderWidth": 0.9,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "center",
              "marginVertical": 10,
              "paddingLeft": 10,
              "width": "100%",
            }
          }
        >
          <Text
            onPress={[Function]}
            style={
              Object {
                "color": "rgba(33, 33, 33, 0.5)",
                "flex": 1,
                "fontSize": 16,
                "fontWeight": "normal",
                "textAlign": "left",
              }
            }
          >
            Select Option
          </Text>
          <Text
            allowFontScaling={false}
            onPress={[Function]}
            style={
              Array [
                Object {
                  "color": "rgba(33, 33, 33, 0.5)",
                  "fontSize": 24,
                },
                Object {
                  "paddingRight": 15,
                },
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
        <Modal
          animationType="fade"
          hardwareAccelerated={false}
          onRequestClose={[Function]}
          supportedOrientations={
            Array [
              "portrait",
              "landscape",
            ]
          }
          transparent={true}
          visible={false}
        >
          <Text
            onPress={[Function]}
            style={
              Object {
                "color": "#000",
                "flex": 1,
                "fontSize": 16,
                "fontWeight": "normal",
                "textAlign": "left",
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "#FFF",
                "flex": 1,
              }
            }
          >
            <RCTPicker
              items={
                Array [
                  Object {
                    "label": "1",
                    "textColor": undefined,
                    "value": "1",
                  },
                  Object {
                    "label": "2",
                    "textColor": undefined,
                    "value": "2",
                  },
                  Object {
                    "label": "3",
                    "textColor": undefined,
                    "value": "3",
                  },
                ]
              }
              onChange={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              selectedIndex={0}
              style={
                Array [
                  Object {
                    "height": 216,
                  },
                  undefined,
                ]
              }
            />
          </View>
          <Text
            onPress={[Function]}
            style={
              Object {
                "color": "#000",
                "flex": 1,
                "fontSize": 16,
                "fontWeight": "normal",
                "textAlign": "left",
              }
            }
          />
        </Modal>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "marginTop": 10,
      }
    }
  >
    <View
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgba(33, 33, 33, 0.87)",
            "fontSize": 15,
            "fontWeight": "normal",
            "marginVertical": 5,
            "textAlign": "left",
          }
        }
      >
        Address
      </Text>
      <View
        style={
          Object {
            "borderColor": "rgba(33, 33, 33, 0.5)",
            "borderRadius": 15,
            "borderWidth": 0.9,
            "marginVertical": 10,
            "paddingHorizontal": 10,
            "paddingTop": 5,
            "width": "100%",
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          multiline={true}
          onBlur={[Function]}
          onContentSizeChange={[Function]}
          onFocus={[Function]}
          style={
            Object {
              "height": 100,
              "width": "100%",
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
    </View>
  </View>
</View>
`;
